sqlserver/tests/conftest.py

```diff
@@ -91,6 +91,13 @@ def instance_ao_docker_secondary():
     return deepcopy(INSTANCE_AO_DOCKER_SECONDARY)
 
 
+@pytest.fixture
+def instance_autodiscovery():
+    instance = INSTANCE_DOCKER
+    instance['database_autodiscovery'] = True
+    return deepcopy(instance)
```

Comment:
```suggestion
    instance = deepcopy(INSTANCE_DOCKER)
    instance['database_autodiscovery'] = True
    return deepcopy(instance)
```
I think `INSTANCE_DOCKER` will be modified without the `deepcopy`

Reply:
Nice catch, this was being done in a few other fixtures as well.